request_logging/middleware.py

```diff
@@ -107,14 +132,14 @@ def __call__(self, request):
         return response
 
     def process_request(self, request):
-        skip_logging, because = self._should_log_route(request)
-        if skip_logging:
-            if because is not None:
-                return self._skip_logging_request(request, because)
+        should_log_route = self._should_log_route(request)
+        if not should_log_route.log_route:
```

Comment:
I realize this mirrors the old code, but why the nested `if`s instead of one `and`? Also, it seems like we should dump out in either case so it should be an `or` (and maybe we provide an empty string if there's no `skip_reason`).